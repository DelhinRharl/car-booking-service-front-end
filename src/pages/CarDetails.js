import React from 'react';
import carImageBig from '../images/car-big.png';

const CarDetails = () => (
  <div className="flex flex-col gap-[30%] p-2 items-center justify-center md:w-4/5">
    <div className="flex  justify-center  w-full">
      <h1 className="font-semibold text-2xl">MERCEDES 2022</h1>
    </div>

    <div className="flex flex-col  gap-[10%] items-center md:flex-row md:gap-[10%] md:items-center">
      <div className="bg-amber-500 w-80 h-80 rounded-full mx-auto relative mb-4 flex justify-center items-center content-center md:bg-amber-500 md:w-[30rem] md:h-[15rem] md:rounded-full ">
        <div className="flex justify-center items-center content-center">
          <img className="w-[100%] h-[100%] md:w-[100%] md:h-[100%]" src={carImageBig} alt="img" />
        </div>
      </div>
      <div className="flex flex-col justify-center items-center gap-4">
        <p className="w-[70%] ">Lorem ipsum, dolor sit amet consectetur adipisicing elit. Error, tempora eligendi! Nesciunt quibusdam autem, numquam velit quo aliquam, porro vero quae recusandae maiores error exercitationem quidem natus voluptatem ab placeat.</p>
        <div className="w-[70%] ">
          <p>Price: $2000/hr</p>
          <p>Fuel: Electric</p>
          <p>Location: Singapore, 12th Lane Rd.</p>
          <p>Contact: +65 98765432</p>
        </div>
        <div>
          <button type="button" className="bg-amber-500    hover:bg-amber-600/80 text-white font-bold py-2 px-4 rounded w-52">
            Reserve
          </button>
        </div>
        <span />
      </div>
    </div>

  </div>
);

export default CarDetails;
