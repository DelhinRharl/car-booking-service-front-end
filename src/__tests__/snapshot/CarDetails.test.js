import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import CarDetails from '../../pages/CarDetails';

describe('CarDetails component', () => {
  it('should render the  page correctly ', () => {
    const header = render(
      <BrowserRouter>
        <CarDetails />
      </BrowserRouter>,
    );

    expect(header).toMatchSnapshot();
  });
});
