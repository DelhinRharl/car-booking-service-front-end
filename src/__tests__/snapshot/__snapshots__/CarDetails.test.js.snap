// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarDetails component should render the  page correctly  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col gap-[30%] p-2 items-center justify-center md:w-4/5"
      >
        <div
          class="flex flex-col gap-[30%]"
        >
          <div
            class="flex  justify-center  w-full"
          >
            <h1
              class="font-semibold text-2xl"
            >
               
              <span>
                 
              </span>
              <br />
              <span />
              <span>
                 
              </span>
               
            </h1>
          </div>
          <div
            class="flex flex-col  gap-[10%] items-center md:flex-row md:gap-[10%] md:items-center"
          >
            <div
              class="bg-amber-500 w-80 h-80 rounded-full mx-auto relative mb-4 flex justify-center items-center content-center md:bg-amber-500 md:w-[30rem] md:h-[15rem] md:rounded-full "
            >
              <div
                class="flex justify-center items-center content-center"
              >
                <img
                  alt="img"
                  class="w-[100%] h-[100%] md:w-[100%] md:h-[100%]"
                />
              </div>
            </div>
            <div
              class="flex flex-col justify-center items-center gap-4"
            >
              <p
                class="w-[70%] "
              >
                Lorem ipsum, dolor sit amet consectetur adipisicing elit. Error, tempora eligendi! Nesciunt quibusdam autem, numquam velit quo aliquam, porro vero quae recusandae maiores error exercitationem quidem natus voluptatem ab placeat.
              </p>
              <div
                class="w-[70%] "
              >
                <p>
                  Price:
                </p>
                <p>
                  Fuel: Electric
                </p>
                <p>
                  Location: Singapore, 12th Lane Rd.
                </p>
                <p>
                  Contact: +65 98765432
                </p>
              </div>
              <div>
                <a
                  href="/reserve"
                >
                  <button
                    class="bg-amber-500    hover:bg-amber-600/80 text-white font-bold py-2 px-4 rounded w-52"
                    type="button"
                  >
                    RESERVE
                  </button>
                </a>
              </div>
              <span />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col gap-[30%] p-2 items-center justify-center md:w-4/5"
    >
      <div
        class="flex flex-col gap-[30%]"
      >
        <div
          class="flex  justify-center  w-full"
        >
          <h1
            class="font-semibold text-2xl"
          >
             
            <span>
               
            </span>
            <br />
            <span />
            <span>
               
            </span>
             
          </h1>
        </div>
        <div
          class="flex flex-col  gap-[10%] items-center md:flex-row md:gap-[10%] md:items-center"
        >
          <div
            class="bg-amber-500 w-80 h-80 rounded-full mx-auto relative mb-4 flex justify-center items-center content-center md:bg-amber-500 md:w-[30rem] md:h-[15rem] md:rounded-full "
          >
            <div
              class="flex justify-center items-center content-center"
            >
              <img
                alt="img"
                class="w-[100%] h-[100%] md:w-[100%] md:h-[100%]"
              />
            </div>
          </div>
          <div
            class="flex flex-col justify-center items-center gap-4"
          >
            <p
              class="w-[70%] "
            >
              Lorem ipsum, dolor sit amet consectetur adipisicing elit. Error, tempora eligendi! Nesciunt quibusdam autem, numquam velit quo aliquam, porro vero quae recusandae maiores error exercitationem quidem natus voluptatem ab placeat.
            </p>
            <div
              class="w-[70%] "
            >
              <p>
                Price:
              </p>
              <p>
                Fuel: Electric
              </p>
              <p>
                Location: Singapore, 12th Lane Rd.
              </p>
              <p>
                Contact: +65 98765432
              </p>
            </div>
            <div>
              <a
                href="/reserve"
              >
                <button
                  class="bg-amber-500    hover:bg-amber-600/80 text-white font-bold py-2 px-4 rounded w-52"
                  type="button"
                >
                  RESERVE
                </button>
              </a>
            </div>
            <span />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
